### ISSUE TITLE
По этому протоколу можно сделать управление?

### ISSUE THREAD

**Issue by dmitryshagin** [doncreat0r/KMENevoBT#2]
Добрый день.
Управляется ли оно в достаточной степени, чтобы сделать переключение газ/бензин по этому протоколу? Идея в том, чтобы сварганить адаптер, который подключить к магнитоле на android (например, через usb+ft232rl), и там сделать простую переключалку газ/бензин в виде иконци на панели. Ну и отображение должно быть поприятнее, чем на этих убогих кнопочках со светодиодами.

**Comment by doncreat0r** [comment]
Привет,
Это можно сделать. Родная программа настройки KME умеет переключать газ/бензин, и делает это по тому же протоколу. Я смутно помню, что вроде бы выяснял, какой именно командой, но реализовывать не стал.

**Comment by dmitryshagin** [comment]
А оно у тебя ещё на машине, или за 4 года всё поменялось?

**Comment by dmitryshagin** [comment]
Я так понял по коду, что это промежуточная софтяра, которая собирает данные с lpg и парктроника(:-O) и отдаёт их в следующий контроллер. Который, подозреваю, гонит их уже куда-то на дисплей. Что-то типа приблуды к colormfa. Фундаментальная работа )

**Comment by doncreat0r** [comment]
Ещё на машине. Но уже глючит, видимо пайка трескается или окисляется. Так что сейчас версия 2 в процессе. Вовремя ты написал, ещё год назад я просто ездил с этой штукой и про внутренности не вспоминал :)

KMENevoBT - собирает данные с KME, китайского парктроника и датчика температуры типа DS18B20 (ещё планируется ловля данных с датчиков давления в шинах). Затем это всё по более простому самодельному протоколу через Bluetooth передаётся на старый смартфон на андроиде 4+. Исходник андроидной софтины тут же на гитхабе, Elpigy который.

**Comment by dmitryshagin** [comment]
У меня TMPS стоит, кто-то даже его отреверсил и выложил протокол: http://4pda.ru/forum/index.php?showtopic=779092&st=2720#entry59838831
Спасибо за труд, я как любитель пореверсить всякую фигню оценил :)
Если вдруг интересно - [мои ковыряния](https://www.drive2.ru/l/480472563638600346/) протокола кнопок с руля и [впиливание блутуса](https://www.drive2.ru/l/10335921/) в штатный мафон

**Comment by doncreat0r** [comment]
Да, я тоже оценил. Ты почто лентопротяг отрубил, кассеты нынче снова в моду входят :)

А TPMS протоколы я прямо с радиоэфира реверсил. Один от чисто китайского комплекта и родной от renault/dacia датчиков. По модуляции они отличаются (OOK и FSK), но есть чипы-приёмники, поддерживающие и ту и другую. С I2C уже идут нужные байты по мере приёма.

В общем, если определю, какой командой газ/бензин переключается, напишу сюда или в прошивку добавлю.
